blocks/subscriptions-block/components/PriceRates/index.jsx

```diff
@@ -0,0 +1,163 @@
+			message = "";
+			break;
+	}
+	return message;
+};
+
+export const ratesUntilCancelled = (billingFrequency) => {
+	let message;
+	switch (billingFrequency) {
+		case DAY:
+			message = "checkout-block.rates-untilCancelled-day";
+			break;
+		case WEEK:
+			message = "checkout-block.rates-untilCancelled-week";
+			break;
+		case MONTH:
+			message = "checkout-block.rates-untilCancelled-month";
+			break;
+		case YEAR:
+			message = "checkout-block.rates-untilCancelled-year";
+			break;
+		default:
+			message = "";
+			break;
+	}
+	return message;
+};
+
+export const ratesDefaultMessage = (billingFrequency, duration) => {
+	const cases = {
+		[DAY]: {
+			[DAY]: "checkout-block.rates-default-day-day",
+			[WEEK]: "checkout-block.rates-default-day-week",
+			[MONTH]: "checkout-block.rates-default-day-month",
+			[YEAR]: "checkout-block.rates-default-day-year",
+		},
+		[WEEK]: {
+			[DAY]: "checkout-block.rates-default-week-day",
+			[WEEK]: "checkout-block.rates-default-week-week",
+			[MONTH]: "checkout-block.rates-default-week-month",
+			[YEAR]: "checkout-block.rates-default-week-year",
+		},
+		[MONTH]: {
+			[DAY]: "checkout-block.rates-default-month-day",
+			[WEEK]: "checkout-block.rates-default-month-week",
+			[MONTH]: "checkout-block.rates-default-month-month",
+			[YEAR]: "checkout-block.rates-default-month-year",
+		},
+		[YEAR]: {
+			[DAY]: "checkout-block.rates-default-year-day",
+			[WEEK]: "checkout-block.rates-default-year-week",
+			[MONTH]: "checkout-block.rates-default-year-month",
+			[YEAR]: "checkout-block.rates-default-year-year",
+		},
+	};
+
+	if (billingFrequency in cases) {
+		if (duration in cases[billingFrequency]) {
+			return cases[billingFrequency][duration];
+		}
+	}
+
+	return undefined;
+};
+
+export const getAmount = (amount, priceCurrency) => `${currency(priceCurrency)}${amount}`;
+
+export const PriceRate = (rate, rateCurrency) => {
+	const phrases = usePhrases();
+
+	const { billingCount, durationCount, billingFrequency, duration } = rate;
+
+	const amount = getAmount(rate?.amount, rateCurrency);
+
+	let rateString = "";
+	let fullRateString = "";
+
+	if (duration === UNTIL_CANCELLED) {
+		if (billingCount <= 1) {
+			rateString = ratesSingleUntilCancelled(billingFrequency);
+			fullRateString = `${amount} ${rateString ? phrases.t(rateString) : ""}`;
+		} else {
+			rateString = ratesUntilCancelled(billingFrequency);
+			fullRateString = `${amount} ${rateString ? phrases.t(rateString, { billingCount }) : ""}`;
+		}
+	}
+
+	if (duration !== UNTIL_CANCELLED) {
+		if (billingCount === durationCount && billingFrequency === duration) {
+			rateString = ratesOneTime(billingFrequency);
+			fullRateString = `${amount} ${rateString ? phrases.t(rateString, { billingCount }) : ""}`;
+		} else {
+			rateString = ratesDefaultMessage(billingFrequency, duration);
+			fullRateString = `${amount} ${rateString ? phrases.t(rateString, { billingCount, durationCount }) : ""}`;
+		}
+	}
+	
+	return fullRateString;
```

Comment:
```suggestion
	if (duration === UNTIL_CANCELLED) {
		if (billingCount <= 1) {
			const rateString = ratesSingleUntilCancelled(billingFrequency);
			return `${amount} ${rateString ? phrases.t(rateString) : ""}`;
		} else {
			const rateString = ratesUntilCancelled(billingFrequency);
			return `${amount} ${rateString ? phrases.t(rateString, { billingCount }) : ""}`;
		}
	} else {
		if (billingCount === durationCount && billingFrequency === duration) {
			const rateString = ratesOneTime(billingFrequency);
			return `${amount} ${rateString ? phrases.t(rateString, { billingCount }) : ""}`;
		} else {
			const rateString = ratesDefaultMessage(billingFrequency, duration);
			return `${amount} ${rateString ? phrases.t(rateString, { billingCount, durationCount }) : ""}`;
		}
	}
```
`let` is a code smell for me.  The initial setting of the values to empty strings is unnecessary as it will never be returned since the two conditions are in opposition (duration is either UNTIL_CANCELLED or it's not.)  It will never return "".

Reply:
@nschubach these updates are causing some linting errors.
<img width="702" alt="Screenshot 2024-05-07 at 5 37 42 p m" src="https://github.com/WPMedia/arc-themes-blocks/assets/54566275/49fd88c5-beda-41d8-907c-3bad412c0f72">

Reply:
Yeah, the else is sort of redundant, TBH.  I should have caught that.  There is no way out of the first if without returning, so the second if was originally redundant... should be:
```
	if (duration === UNTIL_CANCELLED) {
		if (billingCount <= 1) {
			const rateString = ratesSingleUntilCancelled(billingFrequency);
			return `${amount} ${rateString ? phrases.t(rateString) : ""}`;
		} else {
			const rateString = ratesUntilCancelled(billingFrequency);
			return `${amount} ${rateString ? phrases.t(rateString, { billingCount }) : ""}`;
		}
	}

	if (billingCount === durationCount && billingFrequency === duration) {
		const rateString = ratesOneTime(billingFrequency);
		return `${amount} ${rateString ? phrases.t(rateString, { billingCount }) : ""}`;
	} else {
		const rateString = ratesDefaultMessage(billingFrequency, duration);
		return `${amount} ${rateString ? phrases.t(rateString, { billingCount, durationCount }) : ""}`;
	}
```
To avoid the lint error... which changes no logic.  If it's `UNTIL_CANCELLED` it will return inside the block.  If it's not, it will default to the if/else logic in the second part.

Reply:
If you are ok with your update logic, that's fine too.